jib-core/src/main/java/com/google/cloud/tools/jib/configuration/PortsWithProtocol.java

```diff
@@ -79,6 +79,30 @@ public Protocol getProtocol() {
     return protocol;
   }
 
+  @Override
+  public boolean equals(Object other) {
+    if (other == this) {
+      return true;
+    }
+    if (other == null || other.getClass() != PortsWithProtocol.class) {
+      return false;
+    }
+    PortsWithProtocol otherPort = (PortsWithProtocol) other;
+    return minPort == otherPort.minPort
+        && maxPort == otherPort.maxPort
+        && protocol == otherPort.protocol;
+  }
+
+  @Override
+  public int hashCode() {
+    return (protocol == Protocol.TCP ? 1 : -1) * minPort + (maxPort << 16);
```

Comment:
This may work for the assumption that maxPort is within 16 bits, but this class does not guarantee that maxPort is within that size.

Reply:
In that sense, the class does not guarantee non-negative numbers either.